fix spacing & minor tweaks for secrets

davegarrett opened the pull request:
Just a few things I noticed that I think make it a little bit easier to follow.

* Name resumption_secret consistently when used
* Show full legend for each handshake flow
* Simplify/clarify secret sources table a bit and make text cases consistent

The resumption secret naming was also apparently noticed by @martinthomson in issue #191.

(Edited to note: this PR has been pruned down to a smaller list of changes; see commit message)

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -1567,6 +1567,15 @@ ClientKeyShare, as shown in Figure 2:
+               derived from the ephemeral secret.
+
+            [] Indicates messages protected using keys
+               derived from the master secret.
```
I don't think repeating this is helpful

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3152,15 +3176,15 @@ mode (DHE, ECDHE, PSK, etc.) and is summarized in the table below:
 
-    PSK + (EC)DHE               Pre-Shared Key         Client ephemeral
+    PSK + (EC)DHE               pre-shared key         client ephemeral
                                                     w/ server ephemeral
```
I actually think that the caps makes this easier to read.

davegarrett commented:
Ok, to respond to comments I've pushed a new commit.

RE: capitalization: Upper or lower is fine, as long as it's consistent. I've switched it to all upper (title case) instead of all lower.

RE: legend repetition: If you're just reading one section, e.g. got to it via TOC, you might not have seen the legend (recently). I've made it less wordy now and also added a line to indicate "+" is for extensions, as I realized that wasn't necessarily clear. (and added a "+" where the notation was inconsistent) We do have an extension that was previously a message, now. The basic point is consistency; make each handshake flow make sense on its own.

I think repeating the legend is helpful, but if you really don't want to repeat it that I can cut this down to just the little stuff and table. Showing "N/A" instead of repeating things in the table is the most helpful part here, I think. It states that SS=ES in some circumstances elsewhere, but the table isn't clear that these are the exact same values, especially because the extracted ones are _not_.

davegarrett commented:
Edited again and repushed for some more naming consistency.

davegarrett commented:
Revised & rebased. Of note, I think these changes to the table make it far easier to see what's going on here.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -1562,6 +1558,11 @@ ClientKeyShare, as shown in Figure 2:
+          +  extension sent in hello message
+          *  optional/situational and not always sent
+          {} protected using keys derived from ephemeral secret
+          [] protected using keys derived from master secret
```
I still don't think it helps to duplicate these.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3159,37 +3165,41 @@ The exact source of each of these secrets depends on the operational
+    Key Exchange     Static Secret (SS)    Ephemeral Secret (ES)
+    ------------     ------------------    ---------------------
+    1-RTT (EC)DHE       N/A (SS = ES)        Client Ephemeral
+                                           + Server Ephemeral
```
NA isn't correct. You use them both.

davegarrett replied:
It's not applicable in that there is no separate value; it's equal to the other. I guess this would come across better with just "SS = ES". The issue I have with the current version is that it's verbose and not clear that these are generated from the same values. (though, the secrets themselves are different)

The "+" could be a "&" instead.

ekr replied:
SS = ES. xSS != xES.

davegarrett replied:
Yes, that's what I said in the previous comment. They're generated from the same values, but the extracted versions are different. This table here is specifically about the sources of these secrets.

My point here is that the notation & terminology should be more clear and spelled out, as evident by our miscommunications here. ;)

The very short abbreviations may be doing more harm than good. Maybe name SS & ES more explicitly; something like raw_ephemeral_secret?

ekr replied:
I'd like to leave this as-is for now, but maybe I'll clean it up in the rewrite.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3159,37 +3165,41 @@ The exact source of each of these secrets depends on the operational
+  1. static_secret = HKDF(0, SS, "extracted SS", L)
 
-  2. xES = HKDF(0, ES, "extractedES", L)
+  2. ephemeral_secret = HKDF(0, ES, "extracted ES", L)
```
This isn't correct. the static secret is the DH ZZ.

Also, the seeds are a technical change, so remove.

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3251,17 +3261,17 @@ The following table describes the inputs to the key calculation for
+  Application  Master     "application data key expansion"  All handshake
+                                                             messages but
+                                                                 Finished
+                                                           (session_hash)
```
See above.

davegarrett commented:
@ekr: If you're going to be rewriting this, I'll just set those parts aside. The commit here is now just the little stuff: fix the spacing, add a line stating what SS/ES are in the list of derivations, and fix "shared_secret" (there's no such value named that; just drop the underscore).

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3177,7 +3176,10 @@ These shared secret values are used to generate cryptographic keys as
 The derivation process is as follows, where L denotes the length of
-the underlying hash function for HKDF {{RFC5869}}.
+the underlying hash function for HKDF {{RFC5869}} and SS/ES denotes
+the sources from the table above. Whilst SS and ES may be the same
```
SS and ES denote

ekr commented on draft-ietf-tls-tls13.md:
```diff
@@ -3177,7 +3176,10 @@ These shared secret values are used to generate cryptographic keys as
+the underlying hash function for HKDF {{RFC5869}} and SS/ES denotes
+the sources from the table above. Whilst SS and ES may be the same
+in some cases, the extracted static_secret and ephemeral_secret
+will not.
```
xSS and xES

davegarrett commented:
@ekr: commit updated with fixes for your comments above
